allennlp/service/servable/servable.py

```diff
@@ -0,0 +1,59 @@
+from typing import Dict, Any, Optional, List
+
+JSONDict = Dict[str, Any]  # pylint: disable=invalid-name
+
+
+class Servable:
+    """
+    Any "model" that we want to serve through either our command line tool or REST API / webapp
+    needs to implement this interface.
+    """
+    def predict_json(self, inputs: JSONDict) -> JSONDict:
+        raise NotImplementedError()
+
+    # TODO(joelgrus): add a predict_tensor method maybe?
+
+    @classmethod
+    def from_config(cls, config: JSONDict = None) -> 'Servable':  # pylint: disable=unused-argument
+        # default implementation calls no-argument constructor, you probably want to override this
+        return cls()
+
+
+class ServableCollection:
```

Comment:
I could go either way.  The `Registry` is in `allennlp.experiments`, and it's about instantiating various components of models and data.  Serving things is pretty separate, and it makes sense to separate them.  On the other hand, the `Registry` and this `Collection` are trying to solve the same goal, so doing so in a unified way makes sense.